test/e2e/common/utils.py

```diff
@@ -194,16 +194,18 @@ def explain_response(service_name, input_json):
 
 
 def get_cluster_ip(name="istio-ingressgateway", namespace="istio-system"):
-    api_instance = client.CoreV1Api(client.ApiClient())
-    service = api_instance.read_namespaced_service(name, namespace)
-    if service.status.load_balancer.ingress is None:
-        cluster_ip = service.spec.cluster_ip
-    else:
-        if service.status.load_balancer.ingress[0].hostname:
-            cluster_ip = service.status.load_balancer.ingress[0].hostname
+    cluster_ip = os.environ.get("KSERVE_INGRESS_HOST_PORT")
+    if cluster_ip is None:
+        api_instance = client.CoreV1Api(client.ApiClient())
+        service = api_instance.read_namespaced_service(name, namespace)
+        if service.status.load_balancer.ingress is None:
+            cluster_ip = service.spec.cluster_ip
         else:
-            cluster_ip = service.status.load_balancer.ingress[0].ip
-    return os.environ.get("KSERVE_INGRESS_HOST_PORT", cluster_ip)
+            if service.status.load_balancer.ingress[0].hostname:
+                cluster_ip = service.status.load_balancer.ingress[0].hostname
+            else:
+                cluster_ip = service.status.load_balancer.ingress[0].ip
+    return cluster_ip
```

Comment:
How is this different from the original logic?

Reply:
If we set the cluster ip in environment variable, then there is no need to create a client and check for the cluster ip. In previous logic even if we set the environmental variable it goes through the unnecessary steps like creating a client and check for the ingress ip and finally just returning what is set in the env variable.